### PR TITLE
React2 week1

### PR DESCRIPTION
**Pull request by polyovik**
React2 week1 meal-sharing homework

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "scripts": {
     "build": "webpack --mode production",
     "start": "npm run build && node ./src/backend/server",
```
Comment (shpomp) [inline package.json:10]: Hmmmm why is it the whole new app being committed wtih every PR? 🤔

File: src/client/src/api/meals.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  const meals = await axios.get(`${rootHost}/`);
+  
+  return meals.data
+};
+
+const getById = async (id) => {
+  const meals = await axios.get(`${rootHost}/${id}`);
+  
+  return meals.data
+};
```
Comment (shpomp) [inline src/client/src/api/meals.js:14]: Nice and clean! 
However, what if there is an error? 💡

File: src/client/src/components/Meal.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+export default function Meal(params) {
+  const meal = params.meal;
```
Comment (shpomp) [inline src/client/src/components/Meal.jsx]: Again, params is not usual for React props. 
"props" or "meal" and then destructuring the object for its properties, please 🙏🏽

File: src/client/src/components/MealList.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      const res = await mealsAPI.getAll();
+      console.log(res);
+      setMeals(res);
+    }
+    fetchData();
+  }, []);
+
+  return (
+    <div className='meals-container' >
+      {meals.map((meal) => (
```
Comment (shpomp) [inline src/client/src/components/MealList.jsx:22]: Consider the case where meals are not there (yet) 💡

**Review by shpomp — APPROVED** [review]
Very very nice, and some extra stuff too 👏🏽 
A few ideas for improvement, but otherwise very good job! ⭐

File: src/client/src/api/meals.js (hunk 2 above)
Comment (polyovik) [reply src/client/src/api/meals.js:14]: i decided to just use try catch in useEffect:
(`try {
        const res = await mealsAPI.getAll();
        setMeals(res);
      } catch (error) {
        return navigate('/');
      }`)
      
still working on it i think i will change it later

**Comment by polyovik** [comment]
> Very very nice, and some extra stuff too 👏🏽 A few ideas for improvement, but otherwise very good job! ⭐

Thanks for the review 😸  i will improve my code in future commits.

**Pull request merged**